src/containers/SearchPage/SearchContainer.jsx

```diff
@@ -66,7 +66,13 @@ class SearchContainer extends Component {
       key => searchQuery[key] === '' && delete searchQuery[key],
     );
 
-    history.push(toSearch({ ...searchQuery }, type));
+    this.onSearch(searchQuery);
+    this.onUrlChange(searchQuery);
+  }
+
+  onUrlChange(searchQuery) {
+    const { history, type } = this.props;
+    history.push(toSearch(searchQuery, type));
```

Comment:
Hva med å putte search(searchQuery) inn her? Så er den debouncet sammen med url-endringen, og du slipper å ha en egen funksjon for det. Og da re-rendrer ikke komponenten både for søk og for url-endring.